File: GenerateGBFSDataset/src/main/scala/com/david/GenerateGBFS/hadoop/HDFS.scala
package com.david.GenerateGBFS.hadoop

import java.net.URI

import org.apache.hadoop.fs.{FileSystem, Path}
import org.apache.hadoop.conf.Configuration

object HDFS {

  def write(hdfsPath: String, content: String): Unit = {
    val path = new Path(hdfsPath)
    val fs = FileSystem.get(new URI(hdfsPath), getConfiguration())
    if(fs.exists(path)){
      fs.delete(path, true)
    }

    val dataOutputStream = fs.create(path)
    dataOutputStream.write(content.getBytes)
    fs.close()
  }

  def remove(hdfsPath: String): Unit = {
    val path = new Path(hdfsPath)
    val fs = FileSystem.get(new URI(hdfsPath), getConfiguration())
    if(fs.exists(path)){
      fs.delete(path, true)
    }
    fs.close()
  }

  private def getConfiguration(): Configuration = {
    val configuration  = new Configuration()
    configuration.set("fs.hdfs.impl", classOf[org.apache.hadoop.hdfs.DistributedFileSystem].getName)
    configuration.set("fs.file.impl", classOf[org.apache.hadoop.fs.LocalFileSystem].getName)

    return configuration
  }
}

File: Enricher/src/main/java/com/david/enricher/hadoop/HDFS.java
package com.david.enricher.hadoop;

import org.apache.hadoop.conf.Configuration;
import org.apache.hadoop.fs.FileSystem;
import org.apache.hadoop.fs.Path;

import java.io.DataOutputStream;
import java.io.File;
import java.io.IOException;
import java.net.URI;
import java.net.URISyntaxException;

public class HDFS {

    public static void write(String hdfsPath, String content){
        Path path = new Path(hdfsPath);

        try {
            FileSystem fileSystem = FileSystem.get(new URI(hdfsPath), new Configuration());

            DataOutputStream dataOutputStream = fileSystem.create(path);
            dataOutputStream.write(content.getBytes());
            fileSystem.close();
        } catch (IOException e) {
            e.printStackTrace();
        } catch (URISyntaxException e) {
            e.printStackTrace();
        }
    }

    public static boolean move(String hdfsPathSrc, String hdfsPathDst){
        Path pathSrc = new Path(hdfsPathSrc);
        Path pathDst = new Path(hdfsPathDst);
        boolean success = false;

        try{
            FileSystem fileSystem = FileSystem.get(new URI(hdfsPathSrc), new Configuration());

            success = fileSystem.rename(pathSrc, pathDst);
            fileSystem.close();
        } catch (URISyntaxException e) {
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        }

        return success;
    }

    public static void delete(String hdfsPath){
        Path path = new Path(hdfsPath);

        try{
            FileSystem fileSystem = FileSystem.get(new URI(hdfsPath), new Configuration());

            if(fileSystem.exists(path)){
                fileSystem.delete(path, true);
            }

            fileSystem.close();
        } catch (URISyntaxException e) {
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        }

    }
}
